clang/test/SemaSYCL/sycl-device.cpp

```diff
@@ -41,17 +41,13 @@ int* func3() { return nullptr; }
 __attribute__((sycl_device))
 void func3(int *) {}
 #elif defined(WARN_STRICT)
-__attribute__((sycl_device)) // expected-warning {{SYCL 1.2.1 specification does not allow 'sycl_device' attribute applied to a function with a raw pointer return type}}
-int* func3() { return nullptr; }
+__attribute__((sycl_device)) int *func3() { return nullptr; }
```

Comment:
It seems that `NOT_STRICT` and `WARN_STRICT` defines were made to test the diagnostic that is being removed. I think we can remove them as well as corresponding run lines.

Reply:
But otherwise, LGTM.